rioting-daisies/BangazonAPI
# Customer resource CRUD functionality with Tests

# Description

I've created the CustomerController.cs file with all of the regular database methods: Get, Post, Put, Delete. I've implemented the functionality for Query String Parameter 'q'  which returns all resources with matching properties of the input value in the route for `api/Customer?q=[insert value here]`.

I've also created tests for all of the possible return values for these methods. The tests are within the TestCustomers.cs file.

Fixes Issue # [(2)](https://github.com/rioting-daisies/BangazonAPI/issues/2)

## Type of change


# Testing Instructions

1. `git fetch --all`
2. `git checkout cm-startingAPI`
3. In case you've already fetched the branch and it is behind, `git pull origin cm-startingAPI`
4. Review the code in CustomerController.cs
5. Review the code in TestCustomers.cs
6. Run all tests to ensure they all work properly on everyone's machine

# Checklist:
